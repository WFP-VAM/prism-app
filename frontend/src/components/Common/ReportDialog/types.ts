import { Style } from '@react-pdf/types';
import { TFunctionResult } from 'i18next';

export interface PDFLegendDefinition {
  value: string | number | TFunctionResult;
  style: Style | Style[];
}

// This numbers depends on table header and row height
export const MAX_TABLE_ROWS_PER_PAGE = 29;
export const FIRST_PAGE_TABLE_ROWS = 9;
