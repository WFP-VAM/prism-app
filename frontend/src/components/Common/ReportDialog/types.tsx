export enum ReportType {
  Storm,
  Flood,
}
