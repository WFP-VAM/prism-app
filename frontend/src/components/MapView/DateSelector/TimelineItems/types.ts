import {
  CycloneName,
  ShortDate,
  TimeAndState,
} from 'context/anticipatoryAction/AAStormStateSlice/types';

export type DateItemStyle = {
  class: string;
  color: string;
  layerDirectionClass?: string;
  emphasis?: string;
};

export type DateJSON = Record<ShortDate, Record<CycloneName, TimeAndState[]>>;

export type WindStateReport = {
  states: TimeAndState[];
  cycloneName: string | null;
};
