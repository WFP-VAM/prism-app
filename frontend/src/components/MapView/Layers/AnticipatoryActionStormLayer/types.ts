// Q - still needed or needs to be improved?
import { FeatureCollection, Point, Feature } from 'geojson';

export enum FeaturePropertyDataType {
  analysis = 'analysis',
  forecast = 'forecast',
}

interface FeatureProperty {
  data_type: FeaturePropertyDataType;
  time: string;
  development: string;
  // maximum_wind_speed: number;
  // maximum_wind_gust: number;
  // wind_buffer_48: any;
  // wind_buffer_64: any;
}

export interface TimeSeries extends FeatureCollection<Point, FeatureProperty> {}
export interface TimeSeriesFeature extends Feature<Point, FeatureProperty> {}
