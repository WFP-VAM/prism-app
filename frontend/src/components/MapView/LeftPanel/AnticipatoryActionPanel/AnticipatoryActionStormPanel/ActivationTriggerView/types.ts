import { AACategory } from 'context/anticipatoryAction/AAStormStateSlice/parsedStormDataTypes';

export const AAPanelCategories: AACategory[] = [
  AACategory.Severe,
  AACategory.Moderate,
];

export const AADisplayCategory: {
  [key in AACategory]?: string;
} = {
  [AACategory.Severe]: ' >118 KM/H',
  [AACategory.Moderate]: ' >89 KM/H',
};
