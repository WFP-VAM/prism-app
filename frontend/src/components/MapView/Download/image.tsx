import {
  Button,
  CircularProgress,
  createStyles,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { jsPDF } from 'jspdf';
import React from 'react';
import { useSafeTranslation } from '../../../i18n';
import { downloadToFile } from '../utils';

function DownloadImage({
  classes,
  open,
  previewRef,
  handleClose,
}: DownloadImageProps) {
  const { t } = useSafeTranslation();
  const [downloading, setDownloading] = React.useState<boolean>(false);

  const download = async (format: 'pdf' | 'jpeg' | 'png') => {
    const docGeneration = new Promise<void>((resolve, reject) => {
      // png is generally preferred for images containing lines and text.
      const ext = format === 'pdf' ? 'png' : format;
      const canvas = previewRef.current;
      if (!canvas) {
        reject(new Error('canvas is undefined'));
        // return statement to make compiler happy about canvas possibly being undefined
        return;
      }
      const file = canvas.toDataURL(`image/${ext}`);
      if (format === 'pdf') {
        // eslint-disable-next-line new-cap
        const pdf = new jsPDF({
          orientation: 'landscape',
        });
        const imgProps = pdf.getImageProperties(file);
        const pdfWidth = pdf.internal.pageSize.getWidth();
        const pdfHeight = (imgProps.height * pdfWidth) / imgProps.width;
        pdf.addImage(file, 'PNG', 0, 0, pdfWidth, pdfHeight);
        pdf.save('map.pdf');
      } else {
        downloadToFile({ content: file, isUrl: true }, 'map', `image/${ext}`);
      }
      resolve();
    });

    setDownloading(true);
    try {
      await docGeneration;
    } catch (error) {
      console.error(error);
    } finally {
      setDownloading(false);
    }

    handleClose();
  };

  return (
    <Dialog
      maxWidth="xl"
      open={open}
      keepMounted
      onClose={() => handleClose()}
      aria-labelledby="dialog-preview"
    >
      <DialogTitle className={classes.title} id="dialog-preview">
        {t('Map Preview')}
      </DialogTitle>
      <DialogContent>
        <canvas ref={previewRef} />
      </DialogContent>
      <DialogActions>
        <Button onClick={() => handleClose()} color="primary">
          {t('Cancel')}
        </Button>
        <Button
          variant="contained"
          onClick={() => download('png')}
          color="primary"
        >
          {t('Download PNG')}
        </Button>
        <Button
          variant="contained"
          onClick={() => download('jpeg')}
          color="primary"
        >
          {t('Download JPEG')}
        </Button>
        <Button
          variant="contained"
          onClick={() => download('pdf')}
          color="primary"
        >
          {downloading ? (
            <CircularProgress color="secondary" />
          ) : (
            t('Download PDF')
          )}
        </Button>
      </DialogActions>
    </Dialog>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    title: {
      color: theme.palette.text.secondary,
    },
  });

export interface DownloadImageProps extends WithStyles<typeof styles> {
  open: boolean;
  previewRef: React.RefObject<HTMLCanvasElement>;
  handleClose: () => void;
}

export default withStyles(styles)(DownloadImage);
