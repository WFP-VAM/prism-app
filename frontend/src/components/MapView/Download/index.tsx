import React, { useRef, useState } from 'react';
import {
  Button,
  createStyles,
  Grid,
  Hidden,
  ListItemIcon,
  ListItemText,
  MenuItem,
  Theme,
  Tooltip,
  Typography,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import Menu, { MenuProps } from '@material-ui/core/Menu';
import {
  CloudDownload,
  ArrowDropDown,
  Image,
  Description,
} from '@material-ui/icons';
import { useSelector } from 'react-redux';
import {
  layersSelector,
  mapSelector,
} from '../../../context/mapStateSlice/selectors';
import { useSafeTranslation } from '../../../i18n';
import DownloadImage from './image';
import Report from './report';
import { analysisResultSelector } from '../../../context/analysisResultStateSlice';
import { ReportType } from '../utils';

const ExportMenu = withStyles((theme: Theme) => ({
  paper: {
    border: '1px solid #d3d4d5',
    backgroundColor: theme.palette.primary.main,
  },
}))((props: MenuProps) => (
  <Menu
    elevation={0}
    getContentAnchorEl={null}
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'center',
    }}
    transformOrigin={{
      vertical: 'top',
      horizontal: 'center',
    }}
    {...props}
  />
));

const ExportMenuItem = withStyles((theme: Theme) => ({
  root: {
    color: theme.palette.common.white,
  },
}))(MenuItem);

function Download({ classes }: DownloadProps) {
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [openImage, setOpenImage] = useState(false);
  const [openReport, setOpenReport] = useState(false);
  const selectedMap = useSelector(mapSelector);
  const analysisResult = useSelector(analysisResultSelector);
  const selectedLayers = useSelector(layersSelector);
  const previewRef = useRef<HTMLCanvasElement>(null);

  const isShowingStormData = selectedLayers.some(
    ({ id }) => id === 'adamts_buffers',
  );

  const isShowingFloodData = selectedLayers.some(({ id }) =>
    id.includes('hydra_'),
  );

  const shouldShowReport = isShowingFloodData || isShowingStormData;

  const hasAnalysisData = analysisResult !== undefined;

  const tooltipText = (() => {
    if (!shouldShowReport) {
      return 'Reports can only be generated for Tropical Storms or Floods, in hazards menu';
    }
    if (!hasAnalysisData) {
      return 'Run analysis first to generate report';
    }
    return '';
  })();

  const { t } = useSafeTranslation();

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setOpenImage(false);
    setOpenReport(false);
    setAnchorEl(null);
  };

  const openModal = () => {
    if (selectedMap) {
      const activeLayers = selectedMap.getCanvas();
      const canvas = previewRef.current;
      if (canvas) {
        canvas.setAttribute('width', activeLayers.width.toString());
        canvas.setAttribute('height', activeLayers.height.toString());
        const context = canvas.getContext('2d');
        if (context) {
          context.drawImage(activeLayers, 0, 0);
        }
      }
      setOpenImage(true);
    }
    handleClose();
  };

  return (
    <Grid item>
      <Button variant="contained" color="primary" onClick={handleClick}>
        <CloudDownload fontSize="small" />
        <Hidden smDown>
          <Typography className={classes.label} variant="body2">
            {t('Export')}
          </Typography>
        </Hidden>
        <ArrowDropDown fontSize="small" />
      </Button>
      <ExportMenu
        id="export-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        {/* https://v4.mui.com/components/tooltips/#disabled-elements */}
        <Tooltip placement="left" title={tooltipText}>
          <span style={{ display: 'flex' }}>
            <ExportMenuItem
              onClick={() => setOpenReport(true)}
              disabled={!shouldShowReport || !hasAnalysisData}
            >
              <ListItemIcon>
                <Description fontSize="small" htmlColor="white" />
              </ListItemIcon>
              <ListItemText primary={t('REPORT')} />
            </ExportMenuItem>
          </span>
        </Tooltip>

        <ExportMenuItem onClick={openModal}>
          <ListItemIcon>
            <Image fontSize="small" htmlColor="white" />
          </ListItemIcon>
          <ListItemText primary={t('IMAGE')} />
        </ExportMenuItem>
      </ExportMenu>
      <DownloadImage
        open={openImage}
        previewRef={previewRef}
        handleClose={handleClose}
      />
      <Report
        open={openReport}
        handleClose={handleClose}
        reportType={isShowingStormData ? ReportType.Storm : ReportType.Flood}
      />
    </Grid>
  );
}

const styles = () =>
  createStyles({
    label: {
      marginLeft: '10px',
    },
  });

export interface DownloadProps extends WithStyles<typeof styles> {}

export default withStyles(styles)(Download);
