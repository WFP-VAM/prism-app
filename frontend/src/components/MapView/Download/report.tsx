import React from 'react';
import {
  Button,
  createStyles,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
  Theme,
  useTheme,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { useSelector } from 'react-redux';
import { ArrowBack } from '@material-ui/icons';
import { PDFViewer, PDFDownloadLink } from '@react-pdf/renderer';
import { snakeCase } from 'lodash';
import moment from 'moment';
import { useSafeTranslation } from '../../../i18n';
import { mapSelector } from '../../../context/mapStateSlice/selectors';
import {
  analysisResultSelector,
  getCurrentData,
} from '../../../context/analysisResultStateSlice';
import { ReportType } from '../utils';
import { ExposedPopulationResult } from '../../../utils/analysis-utils';
import ReportDoc from './reportDoc';

function Report({ classes, open, reportType, handleClose }: ReportProps) {
  const theme = useTheme();
  const { t } = useSafeTranslation();
  const [mapImage, setMapImage] = React.useState<string | null>(null);
  const selectedMap = useSelector(mapSelector);
  const analysisData = useSelector(getCurrentData);
  const analysisResult = useSelector(
    analysisResultSelector,
  ) as ExposedPopulationResult;
  const eventDate = analysisResult?.date
    ? moment(new Date(analysisResult?.date)).format('YYYY-MM-DD')
    : '';

  const document = (
    <ReportDoc
      t={t}
      exposureLegendDefinition={analysisResult?.legend ?? []}
      theme={theme}
      reportType={reportType}
      tableName="Population Exposure"
      tableShowTotal
      eventName={
        reportType === ReportType.Storm
          ? `Storm Report (${eventDate})`
          : `Flood Report (${eventDate})`
      }
      mapImage={mapImage}
      tableData={analysisData}
    />
  );

  const getPDFName = () => {
    const type = snakeCase(analysisResult?.legendText);
    const date = new Date();
    const dateString = moment(date).format('DD_MM_YYYY');
    return `PRISM_report_${type}_${dateString}.pdf`;
  };

  React.useEffect(() => {
    if (!open) {
      return;
    }
    type Format = 'png' | 'jpeg';
    const getMapImage = (format: Format = 'png'): string | null => {
      if (selectedMap) {
        const activeLayers = selectedMap.getCanvas();
        const file = activeLayers.toDataURL(`image/${format}`);
        return file;
      }
      return null;
    };
    setMapImage(getMapImage('png'));
  }, [open, selectedMap]);

  return (
    <Dialog
      open={open}
      keepMounted
      onClose={() => handleClose()}
      maxWidth={false}
    >
      <DialogTitle className={classes.titleRoot}>
        <div className={classes.title}>
          <IconButton
            className={classes.titleIconButton}
            onClick={() => {
              handleClose();
            }}
          >
            <ArrowBack />
          </IconButton>
          <span className={classes.titleText}>
            {t(
              reportType === ReportType.Storm
                ? 'Storm impact Report'
                : 'Flood Report',
            )}
          </span>
        </div>
      </DialogTitle>
      <DialogContent style={{ height: '90vh', width: 'calc(90vh / 1.42)' }}>
        <div style={{ width: '100%', height: '100%' }}>
          <PDFViewer
            style={{ width: '100%', height: '100%' }}
            showToolbar={false}
          >
            {document}
          </PDFViewer>
        </div>
      </DialogContent>
      <DialogActions className={classes.actions}>
        <span className={classes.signature}>
          {t('P R I S M automated report')}
        </span>
        <Button className={classes.actionButton} variant="outlined">
          <PDFDownloadLink document={document} fileName={getPDFName()}>
            {({ loading }) => (loading ? 'Loading document...' : 'Download')}
          </PDFDownloadLink>
        </Button>
      </DialogActions>
    </Dialog>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    titleRoot: {
      background: theme.dialog?.border,
      padding: 0,
    },
    title: {
      display: 'flex',
      flexWrap: 'nowrap',
      alignItems: 'center',
    },
    titleText: {
      flexGrow: 1,
      textAlign: 'center',
      fontSize: 18,
    },
    titleIconButton: {
      color: '#FFFFFF',
    },
    actions: {
      background: theme.dialog?.border,
      display: 'flex',
      justifyContent: 'space-between',
      flexDirection: 'row',
    },
    actionButton: {
      background: '#FFFFFF',
      color: theme.dialog?.actionButton,
      fontSize: 12,
    },
    signature: {
      fontSize: 12,
      fontWeight: 500,
      paddingLeft: '1em',
    },
  });

export interface ReportProps extends WithStyles<typeof styles> {
  open: boolean;
  reportType: ReportType;
  handleClose: () => void;
}

export default withStyles(styles)(Report);
