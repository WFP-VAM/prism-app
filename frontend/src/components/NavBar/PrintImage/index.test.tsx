import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { Panel } from 'config/types';
import Download from '.';

const mockStore = configureStore([]);

test('renders as expected', () => {
  const realDateNow = Date.now.bind(global.Date);
  const dateNowStub = jest.fn(() => 1530518207007);
  // eslint-disable-next-line fp/no-mutation
  global.Date.now = dateNowStub;

  const store = mockStore({
    mapState: {
      layers: [],
      dateRange: { startDate: 1530518207007 },
      maplibreMap: () => {},
      errors: [],
      layersData: [],
      loadingLayerIds: [],
      boundaryRelationData: {},
    },
    serverState: { availableDates: {}, loading: false },
    anticipatoryActionDroughtState: {
      renderedDistricts: { 'Window 1': {}, 'Window 2': {} },
      filters: { selectedWindow: 'All' },
    },
    analysisResultState: {
      isMapLayerActive: false,
    },
    leftPanelState: {
      tabValue: Panel.AnticipatoryActionDrought,
    },
  });

  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <Download />
      </Provider>
    </BrowserRouter>,
  );
  expect(container).toMatchSnapshot();

  // eslint-disable-next-line fp/no-mutation
  global.Date.now = realDateNow;
});
