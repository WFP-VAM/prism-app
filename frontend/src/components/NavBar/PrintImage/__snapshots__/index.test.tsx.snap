// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    style="padding-top: 4px;"
  >
    <mock-button
      style="background-color: transparent;"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        style="padding-right: 0.2em;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z"
        />
      </svg>
      <mock-typography
        classname="PrintImage-printText-1"
        variant="body2"
      >
        Print
      </mock-typography>
    </mock-button>
  </div>
  <mock-dialog
    aria-labelledby="dialog-preview"
    keepmounted="true"
    maxwidth="xl"
    open="false"
  >
    <mock-dialogtitle
      classname="DownloadImage-title-2"
      id="dialog-preview"
    >
      <div
        style="display: flex; flex-direction: column;"
      >
        Map Preview
        <mock-typography
          color="textSecondary"
          variant="body1"
        >
          Use your mouse to pan and zoom the map
        </mock-typography>
      </div>
      <button
        class="MuiButtonBase-root MuiIconButton-root DownloadImage-closeButton-6"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </mock-dialogtitle>
    <mock-dialogcontent>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
          style="width: 70vw; height: 80vh; margin-bottom: 16px;"
        >
          <div
            style="position: relative; z-index: 3; border: 1px solid black; height: 100%; width: 100%;"
          >
            <div
              class="DownloadImage-printContainer-10"
            >
              <div
                class="DownloadImage-mapOverlay-12"
              />
              <div
                class="DownloadImage-backdropWrapper-8"
              >
                <div
                  aria-hidden="true"
                  class="MuiBackdrop-root DownloadImage-backdrop-7"
                  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
                >
                  <mock-circularprogress />
                </div>
              </div>
              <div
                class="DownloadImage-mapContainer-11"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <div
            class="MuiBox-root MuiBox-root-13"
          >
            <div
              class="MuiBox-root MuiBox-root-14 DownloadImage-title-2"
            >
              Map Options
            </div>
            <div
              class="DownloadImage-toggleWrapper-9"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <mock-switch
                  checked="true"
                  color="primary"
                  name="legend"
                />
                <mock-typography
                  classname="MuiFormControlLabel-label"
                  component="span"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Legend
                  </mock-typography>
                </mock-typography>
              </label>
            </div>
            <div
              class="DownloadImage-toggleWrapper-9"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <mock-switch
                  checked="true"
                  color="primary"
                  name="fullLayerDescription"
                />
                <mock-typography
                  classname="MuiFormControlLabel-label"
                  component="span"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Full Layer Description
                  </mock-typography>
                </mock-typography>
              </label>
            </div>
            <div
              class="DownloadImage-toggleWrapper-9"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <mock-switch
                  checked="true"
                  color="primary"
                  name="footer"
                />
                <mock-typography
                  classname="MuiFormControlLabel-label"
                  component="span"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Footer Text
                  </mock-typography>
                </mock-typography>
              </label>
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="DownloadImage-toggleWrapper-9"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <mock-switch
                  checked="true"
                  color="primary"
                  name="countryMask"
                />
                <mock-typography
                  classname="MuiFormControlLabel-label"
                  component="span"
                >
                  <mock-typography
                    variant="h4"
                  >
                    Country Mask
                  </mock-typography>
                </mock-typography>
              </label>
            </div>
            <mock-typography
              color="textSecondary"
              gutterbottom="true"
              variant="h4"
            >
              Total width
            </mock-typography>
            <span
              class="MuiSlider-root MuiSlider-colorPrimary"
            >
              <span
                class="MuiSlider-rail"
              />
              <span
                class="MuiSlider-track"
                style="left: 0%; width: 100%;"
              />
              <input
                type="hidden"
                value="100"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="0"
                style="left: 0%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="1"
                style="left: 20%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="2"
                style="left: 40%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="3"
                style="left: 60%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="4"
                style="left: 80%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="5"
                style="left: 100%;"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax="100"
                aria-valuemin="50"
                aria-valuenow="100"
                class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index="0"
                role="slider"
                style="left: 100%;"
                tabindex="0"
              />
            </span>
            <mock-typography
              color="textSecondary"
              gutterbottom="true"
              variant="h4"
            >
              Legend scale
            </mock-typography>
            <span
              class="MuiSlider-root MuiSlider-colorPrimary"
            >
              <span
                class="MuiSlider-rail"
              />
              <span
                class="MuiSlider-track"
                style="left: 0%; width: 100%;"
              />
              <input
                type="hidden"
                value="100"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="0"
                style="left: 0%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="1"
                style="left: 20%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="2"
                style="left: 40%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="3"
                style="left: 60%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="4"
                style="left: 80%;"
              />
              <span
                class="MuiSlider-mark MuiSlider-markActive"
                data-index="5"
                style="left: 100%;"
              />
              <span
                aria-orientation="horizontal"
                aria-valuemax="100"
                aria-valuemin="50"
                aria-valuenow="100"
                class="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                data-index="0"
                role="slider"
                style="left: 100%;"
                tabindex="0"
              />
            </span>
            <mock-button
              classname="DownloadImage-gutter-4"
              color="primary"
              endicon="[object Object]"
              variant="contained"
            >
              Download
            </mock-button>
            <mock-menu
              keepmounted="true"
              open="false"
            >
              <mock-menuitem>
                Download PNG
              </mock-menuitem>
              <mock-menuitem>
                Download JPEG
              </mock-menuitem>
              <mock-menuitem>
                Download PDF
              </mock-menuitem>
            </mock-menu>
          </div>
        </div>
      </div>
    </mock-dialogcontent>
  </mock-dialog>
  <mock-dialog
    maxwidth="xl"
    open="false"
  >
    <mock-dialogtitle
      classname="makeStyles-title-20"
    >
      Edit Footer Text
    </mock-dialogtitle>
    <mock-dialogcontent
      style="width: 40rem;"
    >
      <mock-textfield
        fullwidth="true"
        inputprops="[object Object]"
        maxrows="4"
        multiline="true"
        value="Layers represent data on 2018-07-02.  The designations employed and the presentation of material in the map(s) do not imply the expression of any opinion on the part of WFP concerning the legal of constitutional status of any country, territory, city, or sea, or concerning the delimitation of its frontiers or boundaries."
        variant="outlined"
      />
    </mock-dialogcontent>
    <mock-dialogactions>
      <mock-button
        color="secondary"
      >
        Cancel
      </mock-button>
      <mock-button
        color="primary"
      >
        Ok
      </mock-button>
    </mock-dialogactions>
  </mock-dialog>
</div>
`;
